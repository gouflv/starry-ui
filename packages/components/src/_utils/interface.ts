import type { AliasToken } from '@starry-ui/theme'
import { VNode } from 'vue'

export type ComponentToken = AliasToken & {
  componentCls: string
}

export type VNodeTypes = VNode | VNode[]
