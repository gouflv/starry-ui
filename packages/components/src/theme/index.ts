export { css, cx } from '@emotion/css'
export * from './style'
