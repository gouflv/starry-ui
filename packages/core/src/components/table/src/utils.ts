import {
  createColumnHelper,
  type CellContext,
  type Column,
  type ColumnDef,
  type ColumnSizingColumnDef,
  type Header,
  type HeaderGroup
} from '@tanstack/vue-table'
import { get, isObject } from 'lodash-es'
import type { CSSProperties } from 'vue'
import type { ColumnType, CombinedColumnType, GroupColumnType } from '../type'

export function normalizeColumnsKey<R>(
  columns: ColumnType<R>[]
): ColumnType<R>[] {
  const keys: Record<string, boolean> = {}
  return columns.map((column) => {
    let mergedKey = String(column.dataIndex || column.key || '_INNER_COLUMN')
    while (keys[mergedKey]) {
      mergedKey = `${mergedKey}_next`
    }
    keys[mergedKey] = true
    return {
      ...column,
      key: mergedKey
    }
  })
}

export function isGroupColumn<R>(
  column: CombinedColumnType<R>
): column is GroupColumnType<R> {
  return 'children' in column
}

export function crateColumnsDef<R>(
  columns: CombinedColumnType<R>[]
): ColumnDef<R, string>[] {
  const helper = createColumnHelper<R>()
  return columns.map((it, index) => {
    if (isGroupColumn(it)) {
      return helper.group({
        header: it.title,
        columns: it.children.map(transformColumns)
      })
    }
    return transformColumns(it, index)
  })
}

function transformColumns<R>(
  column: ColumnType<R>,
  index: number
): ColumnDef<R, string> {
  const helper = createColumnHelper<R>()

  const unsafeRowKey = `_column-${index}`
  const id = String(column.dataIndex || column.key || unsafeRowKey)

  const cellRender = (ctx: CellContext<R, any>) => {
    if (typeof column.customRender === 'function') {
      return column.customRender({
        text: ctx.getValue(),
        record: ctx.row.original,
        index: ctx.row.index,
        column
      })
    }
    const value = ctx.getValue()
    return isObject(value) ? JSON.stringify(value) : value
  }

  const colSize: ColumnSizingColumnDef = {
    size: column.width,
    minSize: column.minWidth,
    maxSize: column.maxWidth
  }

  if (column.dataIndex) {
    return helper.accessor((r) => get(r, column.dataIndex!), {
      id,
      header: column.title,
      cell: cellRender,
      ...colSize,
      meta: { column }
    })
  }

  return helper.display({
    id,
    header: column.title,
    cell: cellRender,
    ...colSize,
    meta: { column }
  })
}

export function flatColumns<R>(
  columns: CombinedColumnType<R>[]
): ColumnType<R>[] {
  return columns.reduce<ColumnType<R>[]>((acc, column) => {
    if (isGroupColumn(column)) {
      return [...acc, ...flatColumns(column.children)]
    }
    return [...acc, column]
  }, [])
}

export function toSizeValue(num?: number | string) {
  if (typeof num === 'number') {
    return `${num}px`
  }
  if (typeof num === 'string') {
    return num
  }
  return 'auto'
}

/**
 * Calculate the colSpan for the header cell
 *
 * @see https://github.com/TanStack/table/issues/5202#issuecomment-2027529717
 */
export function mergeHeaderGroups<R = any>(
  hg: HeaderGroup<R>[]
): HeaderGroup<R>[] {
  if (hg.length === 1) return hg

  const idCache = new Set<string>()

  return hg.map((group, depth, { length: totalDepth }) => {
    return {
      ...group,
      headers: group.headers
        .filter((hd) => !idCache.has(hd.column.id))
        .map((hd: Header<R, any>) => {
          idCache.add(hd.column.id)
          return hd.isPlaceholder
            ? {
                ...hd,
                isPlaceholder: false,
                rowSpan: totalDepth - depth
              }
            : {
                ...hd,
                rowSpan: 1
              }
        })
    }
  })
}

export function getPinningState(column: Column<any>, componentCls: string) {
  const pinned = column.getIsPinned()
  const isLastLeft = pinned === 'left' && column.getIsLastColumn('left')
  const isFirstRight = pinned === 'right' && column.getIsFirstColumn('right')
  return {
    isFirstRight,
    isLastLeft,
    classes: {
      [`${componentCls}Cell--fixed-left`]: pinned === 'left',
      [`${componentCls}Cell--fixed-right`]: pinned === 'right',
      [`${componentCls}Cell--fixed-left-last`]: isLastLeft,
      [`${componentCls}Cell--fixed-right-first`]: isFirstRight
    },
    style: {
      left: pinned === 'left' ? `${column.getStart('left')}px` : undefined,
      right: pinned === 'right' ? `${column.getAfter('right')}px` : undefined
    } satisfies CSSProperties
  }
}
