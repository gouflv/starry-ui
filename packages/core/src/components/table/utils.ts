import {
  createColumnHelper,
  type CellContext,
  type ColumnDef,
  type ColumnSizingColumnDef,
  type Header,
  type HeaderGroup
} from '@tanstack/vue-table'
import { get, isObject } from 'lodash-es'
import type { ColumnType, CombinedColumnType, GroupColumnType } from './types'

export function normalizeColumnsKey<R>(
  columns: ColumnType<R>[]
): ColumnType<R>[] {
  const keys: Record<string, boolean> = {}
  return columns.map((column) => {
    let mergedKey = String(column.dataIndex || column.key || '_INNER_COLUMN')
    while (keys[mergedKey]) {
      mergedKey = `${mergedKey}_next`
    }
    keys[mergedKey] = true
    return {
      ...column,
      key: mergedKey
    }
  })
}

export function isGroupColumn<R>(
  column: CombinedColumnType<R>
): column is GroupColumnType<R> {
  return 'children' in column
}

export function crateColumnsDef<R>(
  columns: CombinedColumnType<R>[]
): ColumnDef<R, string>[] {
  const helper = createColumnHelper<R>()
  return columns.map((it, index) => {
    if (isGroupColumn(it)) {
      return helper.group({
        header: it.title,
        columns: it.children.map(transformColumns)
      })
    }
    return transformColumns(it, index)
  })
}

function transformColumns<R>(
  column: ColumnType<R>,
  index: number
): ColumnDef<R, string> {
  const helper = createColumnHelper<R>()

  const unsafeRowKey = `_column-${index}`
  const id = String(column.dataIndex || column.key || unsafeRowKey)

  const cellRender = (ctx: CellContext<R, any>) => {
    if (column.customRender) {
      return column.customRender({
        text: ctx.getValue(),
        record: ctx.row.original,
        index: ctx.row.index,
        column
      })
    }
    const value = ctx.getValue()
    if (isObject(value)) {
      return JSON.stringify(value)
    }
    return value
  }

  const colSize: ColumnSizingColumnDef = {
    size: column.width,
    minSize: column.minWidth,
    maxSize: column.maxWidth
  }

  if (column.dataIndex) {
    return helper.accessor((r) => get(r, column.dataIndex!), {
      id,
      header: column.title,
      cell: cellRender,
      ...colSize
    })
  }

  return helper.display({
    id,
    header: column.title,
    cell: cellRender,
    ...colSize
  })
}

export function flatColumns<R>(
  columns: CombinedColumnType<R>[]
): ColumnType<R>[] {
  return columns.reduce<ColumnType<R>[]>((acc, column) => {
    if (isGroupColumn(column)) {
      return [...acc, ...flatColumns(column.children)]
    }
    return [...acc, column]
  }, [])
}

export function toSizeValue(num?: number | string) {
  if (typeof num === 'number') {
    return `${num}px`
  }
  if (typeof num === 'string') {
    return num
  }
  return 'auto'
}

/**
 * @see https://github.com/TanStack/table/issues/5202#issuecomment-2027529717
 */
export function mergeHeaderGroups<R = any>(
  hg: HeaderGroup<R>[]
): HeaderGroup<R>[] {
  if (hg.length === 1) return hg

  const idCache = new Set<string>()

  return hg.map((group, depth, { length: totalDepth }) => {
    return {
      ...group,
      headers: group.headers
        .filter((hd) => !idCache.has(hd.column.id))
        .map((hd: Header<R, any>) => {
          idCache.add(hd.column.id)
          return hd.isPlaceholder
            ? {
                ...hd,
                isPlaceholder: false,
                rowSpan: totalDepth - depth
              }
            : {
                ...hd,
                rowSpan: 1
              }
        })
    }
  })
}
