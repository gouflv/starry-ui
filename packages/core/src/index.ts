export { default as Button } from './components/button'
export { default as Select } from './components/select'
export { default as Table } from './components/table'
export * from './components/theme-token-provider'
