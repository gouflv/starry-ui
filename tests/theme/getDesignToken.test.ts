import { themes } from '@starry/theme'
import { beforeEach, expect, test } from 'vitest'
import getDesignToken from '../../packages/theme/src/cssinjs/utils/getDesignToken'
import { GlobalToken } from '../../packages/theme/src/interface'
import { objectDiff } from '../utils'

let originToken: GlobalToken

beforeEach(() => {
  originToken = getDesignToken()
})

test('should it works', () => {
  expect(originToken).toMatchSnapshot('default designs token')
})

test('should it works with custom seed token', () => {
  const token = getDesignToken({
    token: {
      colorPrimary: 'red'
    }
  })
  expect(token.colorPrimary).toBe('red')
  expect(objectDiff(token, originToken)).toMatchSnapshot('custom seed token')
})

test('should it works with custom seed token 2', () => {
  const token = getDesignToken({
    token: {
      colorPrimary: 'red',
      colorTextBase: 'blue'
    }
  })
  expect(token.colorPrimary).toBe('red')
  expect(token.colorTextBase).toBe('blue')
  expect(objectDiff(token, originToken)).toMatchSnapshot('custom seed token')
})

test('should it works with custom map token', () => {
  const token = getDesignToken({
    token: {
      colorText: 'blue'
    }
  })
  expect(token.colorText).toBe('blue')
  expect(objectDiff(token, originToken)).toMatchSnapshot('custom map token')
})

test('should it works with compact algorithm', () => {
  const token = getDesignToken({
    token: {},
    derivative: [themes.default, themes.compact]
  })
  expect(objectDiff(token, originToken)).toMatchSnapshot('compact algorithm')
})

test('should it works with loosen algorithm', () => {
  const token = getDesignToken({
    token: {},
    derivative: [themes.default, themes.loosen]
  })
  expect(objectDiff(token, originToken)).toMatchSnapshot('loosen algorithm')
})
