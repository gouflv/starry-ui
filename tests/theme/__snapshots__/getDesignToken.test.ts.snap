// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should it works > default designs token 1`] = `
{
  "blue": "#3A8BFF",
  "blue-1": "#f0f9ff",
  "blue-10": "#072266",
  "blue-2": "#def0ff",
  "blue-3": "#b5dbff",
  "blue-4": "#8cc4ff",
  "blue-5": "#63a9ff",
  "blue-6": "#3a8bff",
  "blue-7": "#276bd9",
  "blue-8": "#174eb3",
  "blue-9": "#0b348c",
  "borderRadius": 6,
  "borderRadiusLG": 8,
  "borderRadiusOuter": 4,
  "borderRadiusSM": 4,
  "borderRadiusXS": 2,
  "boxShadow": "
      0 1px 2px 0 rgba(0, 0, 0, 0.03),
      0 1px 6px -1px rgba(0, 0, 0, 0.02),
      0 2px 4px 0 rgba(0, 0, 0, 0.02)
    ",
  "boxShadowCard": "
      0 1px 2px -2px rgba(0, 0, 0, 0.16),
      0 3px 6px 0 rgba(0, 0, 0, 0.12),
      0 5px 12px 4px rgba(0, 0, 0, 0.09)
    ",
  "boxShadowDrawerDown": "
      0 -6px 16px 0 rgba(0, 0, 0, 0.08),
      0 -3px 6px -4px rgba(0, 0, 0, 0.12),
      0 -9px 28px 8px rgba(0, 0, 0, 0.05)
    ",
  "boxShadowDrawerLeft": "
      6px 0 16px 0 rgba(0, 0, 0, 0.08),
      3px 0 6px -4px rgba(0, 0, 0, 0.12),
      9px 0 28px 8px rgba(0, 0, 0, 0.05)
    ",
  "boxShadowDrawerRight": "
      -6px 0 16px 0 rgba(0, 0, 0, 0.08),
      -3px 0 6px -4px rgba(0, 0, 0, 0.12),
      -9px 0 28px 8px rgba(0, 0, 0, 0.05)
    ",
  "boxShadowDrawerUp": "
      0 6px 16px 0 rgba(0, 0, 0, 0.08),
      0 3px 6px -4px rgba(0, 0, 0, 0.12),
      0 9px 28px 8px rgba(0, 0, 0, 0.05)
    ",
  "boxShadowPopoverArrow": "3px 3px 7px rgba(0, 0, 0, 0.1)",
  "boxShadowSecondary": "
      0 6px 16px 0 rgba(0, 0, 0, 0.08),
      0 3px 6px -4px rgba(0, 0, 0, 0.12),
      0 9px 28px 8px rgba(0, 0, 0, 0.05)
    ",
  "boxShadowTabsOverflowBottom": "inset 0 -10px 8px -8px rgba(0, 0, 0, 0.08)",
  "boxShadowTabsOverflowLeft": "inset 10px 0 8px -8px rgba(0, 0, 0, 0.08)",
  "boxShadowTabsOverflowRight": "inset -10px 0 8px -8px rgba(0, 0, 0, 0.08)",
  "boxShadowTabsOverflowTop": "inset 0 10px 8px -8px rgba(0, 0, 0, 0.08)",
  "boxShadowTertiary": "
      0 1px 2px 0 rgba(0, 0, 0, 0.03),
      0 1px 6px -1px rgba(0, 0, 0, 0.02),
      0 2px 4px 0 rgba(0, 0, 0, 0.02)
    ",
  "colorBgBase": "#fff",
  "colorBgContainer": "#ffffff",
  "colorBgContainerDisabled": "rgba(0, 0, 0, 0.04)",
  "colorBgElevated": "#ffffff",
  "colorBgLayout": "#f5f5f5",
  "colorBgMask": "rgba(0, 0, 0, 0.45)",
  "colorBgSpotlight": "rgba(0, 0, 0, 0.85)",
  "colorBgTextActive": "rgba(0, 0, 0, 0.15)",
  "colorBgTextHover": "rgba(0, 0, 0, 0.06)",
  "colorBorder": "#d9d9d9",
  "colorBorderBg": "#ffffff",
  "colorBorderSecondary": "#f0f0f0",
  "colorError": "#ff4d4f",
  "colorErrorActive": "#d9363e",
  "colorErrorBg": "#fff2f0",
  "colorErrorBgHover": "#fff1f0",
  "colorErrorBorder": "#ffccc7",
  "colorErrorBorderHover": "#ffa39e",
  "colorErrorHover": "#ff7875",
  "colorErrorOutline": "rgba(255, 38, 5, 0.06)",
  "colorErrorText": "#ff4d4f",
  "colorErrorTextActive": "#d9363e",
  "colorErrorTextHover": "#ff7875",
  "colorFill": "rgba(0, 0, 0, 0.15)",
  "colorFillAlter": "rgba(0, 0, 0, 0.02)",
  "colorFillContent": "rgba(0, 0, 0, 0.06)",
  "colorFillContentHover": "rgba(0, 0, 0, 0.15)",
  "colorFillQuaternary": "rgba(0, 0, 0, 0.02)",
  "colorFillSecondary": "rgba(0, 0, 0, 0.06)",
  "colorFillTertiary": "rgba(0, 0, 0, 0.04)",
  "colorHighlight": "#ff4d4f",
  "colorIcon": "rgba(0, 0, 0, 0.45)",
  "colorIconHover": "rgba(0, 0, 0, 0.88)",
  "colorInfo": "#3a8bff",
  "colorInfoActive": "#276bd9",
  "colorInfoBg": "#f0f9ff",
  "colorInfoBgHover": "#def0ff",
  "colorInfoBorder": "#b5dbff",
  "colorInfoBorderHover": "#8cc4ff",
  "colorInfoHover": "#8cc4ff",
  "colorInfoText": "#3a8bff",
  "colorInfoTextActive": "#276bd9",
  "colorInfoTextHover": "#63a9ff",
  "colorLink": "#3a8bff",
  "colorLinkActive": "#276bd9",
  "colorLinkHover": "#8cc4ff",
  "colorPrimary": "#3a8bff",
  "colorPrimaryActive": "#276bd9",
  "colorPrimaryBg": "#f0f9ff",
  "colorPrimaryBgHover": "#def0ff",
  "colorPrimaryBorder": "#b5dbff",
  "colorPrimaryBorderHover": "#8cc4ff",
  "colorPrimaryHover": "#63a9ff",
  "colorPrimaryText": "#3a8bff",
  "colorPrimaryTextActive": "#276bd9",
  "colorPrimaryTextHover": "#63a9ff",
  "colorSplit": "rgba(5, 5, 5, 0.06)",
  "colorSuccess": "#52c41a",
  "colorSuccessActive": "#389e0d",
  "colorSuccessBg": "#f6ffed",
  "colorSuccessBgHover": "#d9f7be",
  "colorSuccessBorder": "#b7eb8f",
  "colorSuccessBorderHover": "#95de64",
  "colorSuccessHover": "#95de64",
  "colorSuccessText": "#52c41a",
  "colorSuccessTextActive": "#389e0d",
  "colorSuccessTextHover": "#73d13d",
  "colorText": "rgba(0, 0, 0, 0.88)",
  "colorTextBase": "#000",
  "colorTextDescription": "rgba(0, 0, 0, 0.45)",
  "colorTextDisabled": "rgba(0, 0, 0, 0.25)",
  "colorTextHeading": "rgba(0, 0, 0, 0.88)",
  "colorTextLabel": "rgba(0, 0, 0, 0.65)",
  "colorTextLightSolid": "#fff",
  "colorTextPlaceholder": "rgba(0, 0, 0, 0.25)",
  "colorTextQuaternary": "rgba(0, 0, 0, 0.25)",
  "colorTextSecondary": "rgba(0, 0, 0, 0.65)",
  "colorTextTertiary": "rgba(0, 0, 0, 0.45)",
  "colorWarning": "#faad14",
  "colorWarningActive": "#d48806",
  "colorWarningBg": "#fffbe6",
  "colorWarningBgHover": "#fff1b8",
  "colorWarningBorder": "#ffe58f",
  "colorWarningBorderHover": "#ffd666",
  "colorWarningHover": "#ffd666",
  "colorWarningOutline": "rgba(255, 215, 5, 0.1)",
  "colorWarningText": "#faad14",
  "colorWarningTextActive": "#d48806",
  "colorWarningTextHover": "#ffc53d",
  "colorWhite": "#fff",
  "controlHeight": 32,
  "controlHeightLG": 40,
  "controlHeightSM": 24,
  "controlHeightXS": 16,
  "controlInteractiveSize": 16,
  "controlItemBgActive": "#f0f9ff",
  "controlItemBgActiveDisabled": "rgba(0, 0, 0, 0.15)",
  "controlItemBgActiveHover": "#def0ff",
  "controlItemBgHover": "rgba(0, 0, 0, 0.04)",
  "controlOutline": "rgba(5, 155, 255, 0.06)",
  "controlOutlineWidth": 2,
  "controlPaddingHorizontal": 12,
  "controlPaddingHorizontalSM": 8,
  "controlTmpOutline": "rgba(0, 0, 0, 0.02)",
  "cyan": "#13C2C2",
  "cyan-1": "#e6fffb",
  "cyan-10": "#002329",
  "cyan-2": "#b5f5ec",
  "cyan-3": "#87e8de",
  "cyan-4": "#5cdbd3",
  "cyan-5": "#36cfc9",
  "cyan-6": "#13c2c2",
  "cyan-7": "#08979c",
  "cyan-8": "#006d75",
  "cyan-9": "#00474f",
  "fontFamily": "-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial,
'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol',
'Noto Color Emoji'",
  "fontSize": 14,
  "fontSizeHeading1": 38,
  "fontSizeHeading2": 30,
  "fontSizeHeading3": 24,
  "fontSizeHeading4": 20,
  "fontSizeHeading5": 16,
  "fontSizeIcon": 12,
  "fontSizeLG": 16,
  "fontSizeSM": 12,
  "fontSizeXL": 20,
  "fontWeightStrong": 600,
  "geekblue": "#2F54EB",
  "geekblue-1": "#f0f5ff",
  "geekblue-10": "#030852",
  "geekblue-2": "#d6e4ff",
  "geekblue-3": "#adc6ff",
  "geekblue-4": "#85a5ff",
  "geekblue-5": "#597ef7",
  "geekblue-6": "#2f54eb",
  "geekblue-7": "#1d39c4",
  "geekblue-8": "#10239e",
  "geekblue-9": "#061178",
  "gold": "#FAAD14",
  "gold-1": "#fffbe6",
  "gold-10": "#613400",
  "gold-2": "#fff1b8",
  "gold-3": "#ffe58f",
  "gold-4": "#ffd666",
  "gold-5": "#ffc53d",
  "gold-6": "#faad14",
  "gold-7": "#d48806",
  "gold-8": "#ad6800",
  "gold-9": "#874d00",
  "green": "#52C41A",
  "green-1": "#f6ffed",
  "green-10": "#092b00",
  "green-2": "#d9f7be",
  "green-3": "#b7eb8f",
  "green-4": "#95de64",
  "green-5": "#73d13d",
  "green-6": "#52c41a",
  "green-7": "#389e0d",
  "green-8": "#237804",
  "green-9": "#135200",
  "lime": "#A0D911",
  "lime-1": "#fcffe6",
  "lime-10": "#254000",
  "lime-2": "#f4ffb8",
  "lime-3": "#eaff8f",
  "lime-4": "#d3f261",
  "lime-5": "#bae637",
  "lime-6": "#a0d911",
  "lime-7": "#7cb305",
  "lime-8": "#5b8c00",
  "lime-9": "#3f6600",
  "lineHeight": 1.5714285714285714,
  "lineHeightHeading1": 1.2105263157894737,
  "lineHeightHeading2": 1.2666666666666666,
  "lineHeightHeading3": 1.3333333333333333,
  "lineHeightHeading4": 1.4,
  "lineHeightHeading5": 1.5,
  "lineHeightLG": 1.5,
  "lineHeightSM": 1.6666666666666667,
  "lineType": "solid",
  "lineWidth": 1,
  "lineWidthBold": 2,
  "linkDecoration": "none",
  "linkFocusDecoration": "none",
  "linkHoverDecoration": "none",
  "magenta": "#EB2F96",
  "magenta-1": "#fff0f6",
  "magenta-10": "#520339",
  "magenta-2": "#ffd6e7",
  "magenta-3": "#ffadd2",
  "magenta-4": "#ff85c0",
  "magenta-5": "#f759ab",
  "magenta-6": "#eb2f96",
  "magenta-7": "#c41d7f",
  "magenta-8": "#9e1068",
  "magenta-9": "#780650",
  "margin": 16,
  "marginLG": 24,
  "marginMD": 20,
  "marginSM": 12,
  "marginXL": 32,
  "marginXS": 8,
  "marginXXL": 48,
  "marginXXS": 4,
  "motionBase": 0,
  "motionDurationFast": "0.1s",
  "motionDurationMid": "0.2s",
  "motionDurationSlow": "0.3s",
  "motionUnit": 0.1,
  "orange": "#FA8C16",
  "orange-1": "#fff7e6",
  "orange-10": "#612500",
  "orange-2": "#ffe7ba",
  "orange-3": "#ffd591",
  "orange-4": "#ffc069",
  "orange-5": "#ffa940",
  "orange-6": "#fa8c16",
  "orange-7": "#d46b08",
  "orange-8": "#ad4e00",
  "orange-9": "#873800",
  "padding": 16,
  "paddingContentHorizontal": 12,
  "paddingContentHorizontalLG": 24,
  "paddingContentHorizontalSM": 16,
  "paddingContentVertical": 12,
  "paddingContentVerticalLG": 12,
  "paddingContentVerticalSM": 8,
  "paddingLG": 24,
  "paddingMD": 20,
  "paddingSM": 12,
  "paddingXL": 32,
  "paddingXS": 8,
  "paddingXXS": 4,
  "pink": "#eb2f96",
  "pink-1": "#fff0f6",
  "pink-10": "#520339",
  "pink-2": "#ffd6e7",
  "pink-3": "#ffadd2",
  "pink-4": "#ff85c0",
  "pink-5": "#f759ab",
  "pink-6": "#eb2f96",
  "pink-7": "#c41d7f",
  "pink-8": "#9e1068",
  "pink-9": "#780650",
  "purple": "#722ED1",
  "purple-1": "#f9f0ff",
  "purple-10": "#120338",
  "purple-2": "#efdbff",
  "purple-3": "#d3adf7",
  "purple-4": "#b37feb",
  "purple-5": "#9254de",
  "purple-6": "#722ed1",
  "purple-7": "#531dab",
  "purple-8": "#391085",
  "purple-9": "#22075e",
  "red": "#F5222D",
  "red-1": "#fff1f0",
  "red-10": "#5c0011",
  "red-2": "#ffccc7",
  "red-3": "#ffa39e",
  "red-4": "#ff7875",
  "red-5": "#ff4d4f",
  "red-6": "#f5222d",
  "red-7": "#cf1322",
  "red-8": "#a8071a",
  "red-9": "#820014",
  "screenLG": 992,
  "screenLGMax": 1199,
  "screenLGMin": 992,
  "screenMD": 768,
  "screenMDMax": 991,
  "screenMDMin": 768,
  "screenSM": 576,
  "screenSMMax": 767,
  "screenSMMin": 576,
  "screenXL": 1200,
  "screenXLMax": 1599,
  "screenXLMin": 1200,
  "screenXS": 480,
  "screenXSMax": 575,
  "screenXSMin": 480,
  "screenXXL": 1600,
  "screenXXLMax": 1999,
  "screenXXLMin": 1600,
  "screenXXXL": 2000,
  "screenXXXLMin": 2000,
  "size": 16,
  "sizeLG": 24,
  "sizeMD": 20,
  "sizeMS": 16,
  "sizePopupArrow": 16,
  "sizeSM": 12,
  "sizeStep": 4,
  "sizeUnit": 4,
  "sizeXL": 32,
  "sizeXS": 8,
  "sizeXXL": 48,
  "sizeXXS": 4,
  "volcano": "#FA541C",
  "volcano-1": "#fff2e8",
  "volcano-10": "#610b00",
  "volcano-2": "#ffd8bf",
  "volcano-3": "#ffbb96",
  "volcano-4": "#ff9c6e",
  "volcano-5": "#ff7a45",
  "volcano-6": "#fa541c",
  "volcano-7": "#d4380d",
  "volcano-8": "#ad2102",
  "volcano-9": "#871400",
  "yellow": "#FADB14",
  "yellow-1": "#feffe6",
  "yellow-10": "#614700",
  "yellow-2": "#ffffb8",
  "yellow-3": "#fffb8f",
  "yellow-4": "#fff566",
  "yellow-5": "#ffec3d",
  "yellow-6": "#fadb14",
  "yellow-7": "#d4b106",
  "yellow-8": "#ad8b00",
  "yellow-9": "#876800",
  "zIndexBase": 0,
  "zIndexPopupBase": 1000,
}
`;

exports[`should it works with compact algorithm > compact algorithm 1`] = `
{
  "controlHeight": 28,
  "controlHeightLG": 35,
  "controlHeightSM": 21,
  "controlHeightXS": 14,
  "controlInteractiveSize": 14,
  "fontSize": 12,
  "fontSizeHeading1": 32,
  "fontSizeHeading2": 26,
  "fontSizeHeading3": 20,
  "fontSizeHeading4": 16,
  "fontSizeHeading5": 14,
  "fontSizeIcon": 10,
  "fontSizeLG": 14,
  "fontSizeSM": 10,
  "fontSizeXL": 16,
  "lineHeight": 1.6666666666666667,
  "lineHeightHeading1": 1.25,
  "lineHeightHeading2": 1.3076923076923077,
  "lineHeightHeading3": 1.4,
  "lineHeightHeading4": 1.5,
  "lineHeightHeading5": 1.5714285714285714,
  "lineHeightLG": 1.5714285714285714,
  "lineHeightSM": 1.8,
  "margin": 8,
  "marginLG": 16,
  "marginMD": 16,
  "marginSM": 8,
  "marginXS": 4,
  "padding": 8,
  "paddingContentHorizontal": 8,
  "paddingContentHorizontalLG": 16,
  "paddingContentHorizontalSM": 8,
  "paddingContentVertical": 8,
  "paddingContentVerticalLG": 8,
  "paddingContentVerticalSM": 4,
  "paddingLG": 16,
  "paddingMD": 16,
  "paddingSM": 8,
  "paddingXS": 4,
  "size": 8,
  "sizeLG": 16,
  "sizeMD": 16,
  "sizeMS": 12,
  "sizeSM": 8,
  "sizeXS": 4,
}
`;

exports[`should it works with custom map token > custom map token 1`] = `
{
  "colorIconHover": "blue",
  "colorText": "blue",
  "colorTextHeading": "blue",
}
`;

exports[`should it works with custom seed token > custom seed token 1`] = `
{
  "colorPrimary": "red",
  "colorPrimaryActive": "#d90007",
  "colorPrimaryBg": "#ffeae6",
  "colorPrimaryBgHover": "#ffafa3",
  "colorPrimaryBorder": "#ff887a",
  "colorPrimaryBorderHover": "#ff5d52",
  "colorPrimaryHover": "#ff3029",
  "colorPrimaryText": "#ff0000",
  "colorPrimaryTextActive": "#d90007",
  "colorPrimaryTextHover": "#ff3029",
  "controlItemBgActive": "#ffeae6",
  "controlItemBgActiveHover": "#ffafa3",
  "controlOutline": "rgba(255, 45, 5, 0.1)",
}
`;

exports[`should it works with custom seed token 2 > custom seed token 1`] = `
{
  "colorBgContainerDisabled": "rgba(0, 0, 255, 0.04)",
  "colorBgSpotlight": "rgba(0, 0, 255, 0.85)",
  "colorBgTextActive": "rgba(0, 0, 255, 0.15)",
  "colorBgTextHover": "rgba(0, 0, 255, 0.06)",
  "colorFill": "rgba(0, 0, 255, 0.15)",
  "colorFillAlter": "rgba(0, 0, 255, 0.02)",
  "colorFillContent": "rgba(0, 0, 255, 0.06)",
  "colorFillContentHover": "rgba(0, 0, 255, 0.15)",
  "colorFillQuaternary": "rgba(0, 0, 255, 0.02)",
  "colorFillSecondary": "rgba(0, 0, 255, 0.06)",
  "colorFillTertiary": "rgba(0, 0, 255, 0.04)",
  "colorIcon": "rgba(0, 0, 255, 0.45)",
  "colorIconHover": "rgba(0, 0, 255, 0.88)",
  "colorPrimary": "red",
  "colorPrimaryActive": "#d90007",
  "colorPrimaryBg": "#ffeae6",
  "colorPrimaryBgHover": "#ffafa3",
  "colorPrimaryBorder": "#ff887a",
  "colorPrimaryBorderHover": "#ff5d52",
  "colorPrimaryHover": "#ff3029",
  "colorPrimaryText": "#ff0000",
  "colorPrimaryTextActive": "#d90007",
  "colorPrimaryTextHover": "#ff3029",
  "colorText": "rgba(0, 0, 255, 0.88)",
  "colorTextBase": "blue",
  "colorTextDescription": "rgba(0, 0, 255, 0.45)",
  "colorTextDisabled": "rgba(0, 0, 255, 0.25)",
  "colorTextHeading": "rgba(0, 0, 255, 0.88)",
  "colorTextLabel": "rgba(0, 0, 255, 0.65)",
  "colorTextPlaceholder": "rgba(0, 0, 255, 0.25)",
  "colorTextQuaternary": "rgba(0, 0, 255, 0.25)",
  "colorTextSecondary": "rgba(0, 0, 255, 0.65)",
  "colorTextTertiary": "rgba(0, 0, 255, 0.45)",
  "controlItemBgActive": "#ffeae6",
  "controlItemBgActiveDisabled": "rgba(0, 0, 255, 0.15)",
  "controlItemBgActiveHover": "#ffafa3",
  "controlItemBgHover": "rgba(0, 0, 255, 0.04)",
  "controlOutline": "rgba(255, 45, 5, 0.1)",
  "controlTmpOutline": "rgba(0, 0, 255, 0.02)",
}
`;

exports[`should it works with loosen algorithm > loosen algorithm 1`] = `
{
  "controlHeight": 36,
  "controlHeightLG": 45,
  "controlHeightSM": 27,
  "controlHeightXS": 18,
  "controlInteractiveSize": 18,
  "fontSize": 16,
  "fontSizeHeading1": 42,
  "fontSizeHeading2": 34,
  "fontSizeHeading3": 28,
  "fontSizeHeading4": 22,
  "fontSizeHeading5": 18,
  "fontSizeIcon": 14,
  "fontSizeLG": 18,
  "fontSizeSM": 14,
  "fontSizeXL": 22,
  "lineHeight": 1.5,
  "lineHeightHeading1": 1.1904761904761905,
  "lineHeightHeading2": 1.2352941176470589,
  "lineHeightHeading3": 1.2857142857142858,
  "lineHeightHeading4": 1.3636363636363635,
  "lineHeightHeading5": 1.4444444444444444,
  "lineHeightLG": 1.4444444444444444,
  "lineHeightSM": 1.5714285714285714,
  "margin": 24,
  "marginLG": 32,
  "marginMD": 32,
  "marginSM": 24,
  "marginXL": 48,
  "marginXS": 20,
  "marginXXL": 64,
  "marginXXS": 20,
  "padding": 24,
  "paddingContentHorizontal": 24,
  "paddingContentHorizontalLG": 32,
  "paddingContentHorizontalSM": 24,
  "paddingContentVertical": 24,
  "paddingContentVerticalLG": 24,
  "paddingContentVerticalSM": 20,
  "paddingLG": 32,
  "paddingMD": 32,
  "paddingSM": 24,
  "paddingXL": 48,
  "paddingXS": 20,
  "paddingXXS": 20,
  "size": 24,
  "sizeLG": 32,
  "sizeMD": 32,
  "sizeMS": 28,
  "sizeSM": 24,
  "sizeXL": 48,
  "sizeXS": 20,
  "sizeXXL": 64,
  "sizeXXS": 20,
}
`;
